utils/blackbox_test.py

```diff
@@ -0,0 +1,97 @@
+from itertools import product
+from pathlib import Path
+import pytest
+
+import pyteal as pt
+
+from utils.blackbox import blackbox_pyteal
+
+from tests.compile_asserts import assert_teal_as_expected
+
+TEAL_PATH = Path.cwd() / "utils" / "test" / "teal"
+
+
+@pt.Subroutine(pt.TealType.none, input_types=[])
+def utest_noop():
+    return pt.Pop(pt.Int(0))
+
+
+@pt.Subroutine(
+    pt.TealType.none,
+    input_types=[pt.TealType.uint64, pt.TealType.bytes, pt.TealType.anytype],
+)
+def utest_noop_args(x, y, z):
+    return pt.Pop(pt.Int(0))
+
+
+@pt.Subroutine(pt.TealType.uint64, input_types=[])
+def utest_int():
+    return pt.Int(0)
+
+
+@pt.Subroutine(
+    pt.TealType.uint64,
+    input_types=[pt.TealType.uint64, pt.TealType.bytes, pt.TealType.anytype],
+)
+def utest_int_args(x, y, z):
+    return pt.Int(0)
+
+
+@pt.Subroutine(pt.TealType.bytes, input_types=[])
+def utest_bytes():
+    return pt.Bytes("")
+
+
+@pt.Subroutine(
+    pt.TealType.bytes,
+    input_types=[pt.TealType.uint64, pt.TealType.bytes, pt.TealType.anytype],
+)
+def utest_bytes_args(x, y, z):
+    return pt.Bytes("")
+
+
+@pt.Subroutine(pt.TealType.anytype, input_types=[])
+def utest_any():
+    x = pt.ScratchVar(pt.TealType.anytype)
+    return pt.Seq(x.store(pt.Int(0)), x.load())
+
+
+@pt.Subroutine(
+    pt.TealType.anytype,
+    input_types=[pt.TealType.uint64, pt.TealType.bytes, pt.TealType.anytype],
+)
+def utest_any_args(x, y, z):
+    x = pt.ScratchVar(pt.TealType.anytype)
+    return pt.Seq(x.store(pt.Int(0)), x.load())
+
+
+UNITS = [
+    utest_noop,
+    utest_noop_args,
+    utest_int,
+    utest_int_args,
+    utest_bytes,
+    utest_bytes_args,
+    utest_any,
+    utest_any_args,
+]
+
+
+@pytest.mark.parametrize("subr, mode", product(UNITS, pt.Mode))
+def test_blackbox_pyteal(subr, mode):
+    """
+    TODO: here's an example of issue #199 at play - (the thread-safety aspect):
+    compare the following!
+    % pytest -n 2 tests/integration/graviton_test.py::test_blackbox_pyteal
+    vs
+    % pytest -n 1 tests/integration/graviton_test.py::test_blackbox_pyteal
+    """
+    is_app = mode == pt.Mode.Application
+    name = f"{'app' if is_app else 'lsig'}_{subr.name()}"
+
+    compiled = pt.compileTeal(blackbox_pyteal(subr, mode)(), mode, version=6)
+    save_to = TEAL_PATH / (name + ".teal")
```

Comment:
@tzaffi Several thoughts here around _hopefully_ helping maintainers understand repo structure.

If agreed to, I am on-board to make the changes in a follow-on PR if preferred.

* Can `tests/teal` be moved to `tests/unit/teal` to help demarcate its usage?  On the surface, it's unclear what will generate output to the dir.
* So that we can treat `teal/` dirs as generated-only dirs, are you opposed to keeping the `_expected.teal` in a different dir than `tests/teal`?
  * Rationale: The dir is cluttered with generated files.  It's hard to discern which files _should_ be retained.  Currently, most `_expected.teal` files are no longer referenced and can be deleted.
  * For example:  To locate closer to usage, can be moved to `tests/integration/resources` with an analogous generated dir `tests/integration/teal`.

Reply:
This is now implemented

Reply:
@tzaffi Thanks for the changes to bring in a `generated` dir and restructure `tests/`.  ☕ 

Looks good to me as of https://github.com/algorand/pyteal/pull/249/commits/8adec2624225788d63763a0259f812bedd6eb0f4.  Resolving.